ddtrace/appsec/_patch_subprocess_executions.py

```diff
@@ -0,0 +1,435 @@
+        trace_utils.wrap(os, "fork", traced_fork(os))
+
+        # all os.spawn* variants eventually use this one:
+        trace_utils.wrap(os, "_spawnvef", traced_osspawn(os))
+
+        if PY2:
+            # note: popen* uses subprocess in Python3, which we already wrap below, but not in
+            # Python2
+            trace_utils.wrap(os, "popen", traced_py2popen(os))
+            trace_utils.wrap(os, "popen2", traced_py2popen(os))
+            trace_utils.wrap(os, "popen3", traced_py2popen(os))
+            trace_utils.wrap(os, "popen4", traced_py2popen(os))
+        patched.append("os")
+
+    if not getattr(subprocess, "_datadog_patch", False):
+        Pin().onto(subprocess)
+        # We store the parameters on __init__ in the context and set the tags on wait
+        # (where all the Popen objects eventually arrive, unless killed before it)
+        trace_utils.wrap(subprocess, "Popen.__init__", traced_subprocess_init(subprocess))
+        trace_utils.wrap(subprocess, "Popen.wait", traced_subprocess_wait(subprocess))
+
+        setattr(os, "_datadog_patch", True)
+        setattr(subprocess, "_datadog_patch", True)
+        patched.append("subprocess")
+
+    return patched
+
+
+@attr.s(eq=False)
+class SubprocessCmdLineCacheEntry(object):
+    binary = attr.ib(type=str, default=None)
+    arguments = attr.ib(type=List, default=None)
+    truncated = attr.ib(type=bool, default=False)
+    env_vars = attr.ib(type=List, default=None)
+    as_list = attr.ib(type=List, default=None)
+    as_string = attr.ib(type=str, default=None)
+
+
+class SubprocessCmdLine(object):
+    # This catches the computed values into a SubprocessCmdLineCacheEntry object
+    _CACHE = {}  # type: Dict[str, SubprocessCmdLineCacheEntry]
+    _CACHE_DEQUE = collections.deque()  # type: Deque[str]
+    _CACHE_MAXSIZE = 32
+    _CACHE_LOCK = RLock()
+
+    @classmethod
+    def _add_new_cache_entry(cls, key, env_vars, binary, arguments, truncated):
+        if key in cls._CACHE:
+            return
+
+        cache_entry = SubprocessCmdLineCacheEntry()
+        cache_entry.binary = binary
+        cache_entry.arguments = arguments
+        cache_entry.truncated = truncated
+        cache_entry.env_vars = env_vars
+
+        with cls._CACHE_LOCK:
+            if len(cls._CACHE_DEQUE) >= cls._CACHE_MAXSIZE:
+                # If the cache is full, remove the oldest entry
+                last_cache_key = cls._CACHE_DEQUE[-1]
+                del cls._CACHE[last_cache_key]
+                cls._CACHE_DEQUE.pop()
+
+            cls._CACHE[key] = cache_entry
+            cls._CACHE_DEQUE.appendleft(key)
+
+        return cache_entry
+
+    @classmethod
+    def _clear_cache(cls):
+        with cls._CACHE_LOCK:
+            cls._CACHE_DEQUE.clear()
+            cls._CACHE.clear()
+
+    TRUNCATE_LIMIT = 4 * 1024
+
+    ENV_VARS_ALLOWLIST = {"LD_PRELOAD", "LD_LIBRARY_PATH", "PATH"}
+
+    BINARIES_DENYLIST = {
+        "md5",
+    }
+
+    SENSITIVE_WORDS_WILDCARDS = (
```

Comment:
Can we make this list configurable via environment variable? I can see a scenario where customers would like to add sensitive words. 

This is not a blocking comment, we can do this in a follow up

Reply:
Since I think this is a great suggestion and the current RFC wording doesn't explicitly disallow to add more words (it disallow removing them) I'll implement this.

Reply:
Commited, PTAL when you can.